tests/functional/adapter/test_detailed_table_type.py

```diff
@@ -0,0 +1,46 @@
+import re
+
+import pytest
+
+from dbt.tests.util import run_dbt, run_dbt_and_capture
+
+get_detailed_table_type_sql = """
```

Comment:
I'm wondering why you went for this approach instead of calling `{% macro athena__list_relations_without_caching(schema_relation) %}` inside a model, and checking that distinct detailed_table_type is iceberg. Imho a bit easier and more helpful to test, as we check an actual macro available in the adapter. Wdyt?

Reply:
I used `list_relations_without_caching` macro inside another macro, because I can't understand how I can get macro results in model and than assert it in test
Seems to me it would be easier. I can try to implement it.

Reply:
Give it a try, if doesn't work we could also leave this implementation

Reply:
I've tried to execute `list_relations_without_caching` macro inside model for the test model to retrieve Athena Relation, but it return empty list instead of list with relation for the test model. 
I think, we should leave my current implementation for functional test as is 🤔

Reply:
Did you pass to list_relations_without_caching the schema used by the tests?

see [here](https://github.com/dbt-athena/dbt-athena/blob/e6678691b37390407e7b369b6f3c2277d5b4a720/tests/functional/adapter/test_force_batch.py#L121) how can reference test schema.

Reply:
Here's a revised version of your text with corrected grammar and improved clarity:

I passed `this.schema` inside the model, which has the same name as the `project.test_schema` reference. However, the model did not run in non-execute mode. When I run the model in execute mode, the macro list_relations_without_caching within the model returns an empty list. In contrast, in another test macros in execute mode, list_relations_without_caching returns the list that I expect.

Reply:
looks good, let's leave **get_detailed_table_type**  macro